src/fsharp/build.fs

```diff
@@ -2705,9 +2705,15 @@ type TcConfig private (data : TcConfigBuilder,validate:bool) =
         let isNetModule = String.Compare(ext,".netmodule",StringComparison.OrdinalIgnoreCase)=0 
         if String.Compare(ext,".dll",StringComparison.OrdinalIgnoreCase)=0 
            || String.Compare(ext,".exe",StringComparison.OrdinalIgnoreCase)=0 
-           || isNetModule then 
+           || isNetModule then
 
-            let resolved = TryResolveFileUsingPaths(tcConfig.SearchPathsForLibraryFiles,m,nm)
+            let searchPaths =
+                if m <> range0 && m <> rangeStartup && m <> rangeCmdArgs && FileSystem.IsPathRootedShim m.FileName then
+                    tcConfig.SearchPathsForLibraryFiles @ [Path.GetDirectoryName(m.FileName)]
```

Comment:
As related to this bug, the search paths did not previously contain the directory of the source file declaring the `#r` reference.  The range is passed to `TryResolveFileUsingPaths` but not actually used.

With this change, we make sure that the directory of the source file is always included in the search path list.

I've added it to the *end* of the search list so that any previous successful resolution will behave the same, it's only previously-failing resolutions that now have an extra chance to succeed.